packages/frontend/core/src/layouts/workspace-layout.tsx

```diff
@@ -116,10 +117,48 @@ type WorkspaceLayoutProps = {
   migration?: MigrationPoint;
 };
 
+const useSyncWorkspaceBlob = () => {
+  // temporary solution for sync blob
+
+  const [currentWorkspace] = useCurrentWorkspace();
+
+  useEffect(() => {
+    const blobEngine = getBlobEngine(currentWorkspace.blockSuiteWorkspace);
+    let stopped = false;
+    function sync() {
+      if (stopped) {
+        return;
+      }
+
+      blobEngine
+        ?.sync()
+        .catch(error => {
+          console.error('sync blob error', error);
+        })
+        .finally(() => {
+          // sync every 1 minute
+          setTimeout(() => {
+            sync();
+          }, 60000);
+        });
+    }
+
+    // after currentWorkspace changed, wait 1 second to start sync
+    setTimeout(() => {
+      sync();
+    }, 1000);
+
+    return () => {
+      stopped = true;
```

Comment:
why don't clear timeout here

Reply:
Because clear timeout is useless when running the promise

```
 -> promise
   .finish(setTimeout)
```

We still need a variable

Reply:
If not cleared, it seems that an increasing number of recursive calls will accumulate.

Reply:
Recursion will be interrupted when `stopped == true`